src/kw_config_loader.sh

```diff
@@ -159,11 +183,11 @@ function load_configuration()
 {
   local -a config_dirs
   local config_dirs_size
-  local IFS=:
-  read -ra config_dirs <<< "${XDG_CONFIG_DIRS:-"/etc/xdg"}"
-  unset IFS
+
+  IFS=: read -ra config_dirs <<< "${XDG_CONFIG_DIRS:-"/etc/xdg"}"
 
   parse_configuration "$KW_ETC_DIR/$CONFIG_FILENAME"
+  parse_configuration "$KW_ETC_DIR/$BUILD_CONFIG_FILENAME" build_config
```

Comment:
What if we move the `build.config` load to `src/build.sh`? In the current implementation, we are loading all config every time that we include this file (`kw_cionfig_loader`). Maybe we can make the load option per feature and save some CPU cycles.

Reply:
took me a while to do this, and I think some tests didn't end up as they were supposed to. Maybe if we could merge the test fixing PR before this one it'd help in this review.